### PR TITLE
fix doc rendering issues, part I

### PR DESCRIPTION
**Pull request by hoefling**
Signed-off-by: Oleg Hoefling <oleg.hoefling@gmail.com>

## Description

This PR fixes wrong placement of the `conf-database-backend` ref label, which prevents hyperlinks from rendering, e.g. [here](https://docs.celeryq.dev/en/latest/userguide/configuration.html#result-backend):

image

### CODE REVIEW

**Comment by auvipy** [comment]
no need to open so many such PR. you can address all of them in single PR

**Comment by hoefling** [comment]
@auvipy gladly, but beware the diff might get a bit heavy. Let me ping you when I'm done with the PR.

**Comment by auvipy** [comment]
thats not big enough actually

**Comment by hoefling** [comment]
@auvipy there's more to do actually, but let's submit the changes in chunks - otherwise it'll be a PITA to review. Thus requesting a review 😎

File: docs/reference/celery.bin.amqp.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+====================
```
Comment (auvipy) [inline docs/reference/celery.bin.amqp.rst:1]: was this removed mistakenly?

**Review by auvipy — APPROVED** [review]

**Pull request merged**